### PR TITLE
Cocoapods automation

### PR DESCRIPTION
**Pull request by TomRHandcock**
- Integrates Cocoapods Gradle plugin with project
- Adds Podspec generation to `publish_swift_package` Fastlane lane

### CODE REVIEW

File: fastlane/fastfiles/lib.rb [hunk 1]
```diff
@@ -64,6 +78,9 @@
                 git tag v#{get_version_name}
                 git push --tags
             })
+
+            # Push podspec to trunk
+            sh("pod trunk push")
```
Comment (SamC-Apadmi) [inline fastlane/fastfiles/lib.rb:83]: Was just about to merge this then it occured to me, should we skip this on Snapshot builds?

Comment (TomRHandcock) [reply fastlane/fastfiles/lib.rb:83]: Ah good spot, that was my intention but looks like I screwed up resolving a merge conflict. Will make a patch when I get a chance 👍

Comment (TomRHandcock) [reply fastlane/fastfiles/lib.rb:83]: @SamC-Apadmi Was about to fix this however this command is within the `else` block. As such it looks to me this is being skipped on snapshot builds (?)

Comment (SamC-Apadmi) [reply fastlane/fastfiles/lib.rb:83]: Good point!

**Review by SamC-Apadmi — APPROVED** [review]

**Pull request merged**